### PR TITLE
[DF-63] Add display names to components

### PR DESCRIPTION
**Pull request by danidewitt**
## What did we change?
- Added display names to `Ez-` components

## Why are we doing this?
- Assist with debugging, closes https://github.com/ezcater/recipe/issues/196

## Checklist

- [ ] Provide test coverage for the changes made
- [x] Create a changeset (`npm run changeset`) with a summary of the changes

[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

**Comment by danidewitt** [comment]
@CraigCav I think the only one I didn't add a display name to was `EzItem` - bc [prop types](https://github.com/ezcater/recipe/blob/e8adbf10c77dab394e5c597bce747d7ce20a4410/src/components/EzContent/EzItem.tsx#L4).

**Comment by CraigCav** [comment]
@danidewitt The main issue with https://github.com/ezcater/recipe/issues/196, is with default exports with no name:

```jsx
// 🛑  - default export is an anonymous function, so React can't infer a name
export default () => {
  return <button />;
}
```

Any of the following patterns will show up named in DevTools and warnings, without needing an explicit `displayName` added:

```jsx
// ✅ - Named export (function declaration)
export function Button() {
  return <button />;
}

// ✅ - Named export (arrow function)
export const Button = () =>  {
  return <button />;
};

// ✅ - Default export  (function declaration)
export default function Button() {
  return <button />;
}

// ✅ - Default export (arrow function - dev tools picks up the function's `.name`, which is inferred from the variable name)
const Button = () =>  {
  return <button />;
};

export default Button;
```

I **_think_** that in the majority of cases, we follow the last example (Default export of a named arrow function), but there may be some components that don't follow that pattern and slip through the cracks. We should be able to find those cases by searching for `export default () =>`. I'd prefer to avoid blanket adding `displayName` across the board, as we're then needlessly shipping extra bytes.

So it's really just `export default () =>` we have to worry about. Those:

- show up as Unknown in React dev tools
- show up as anonymous in stack traces
- won't be checked by react's linting rules (i.e. checking for `use{Hookname}` etc)
- won't work with certain react features, like Fast Refresh.

File: src/components/EzCard/EzCardHeading.tsx [hunk 1]
```diff
@@ -39,6 +39,6 @@ const EzCardHeading: React.FC<HeadingProps> = ({title, subtitle, actions}) => {
   return <div className={box()}>{actions ? layout : heading}</div>;
 };
 
-EzCardHeading.displayName = 'EzCard';
+EzCardHeading.displayName = 'EzCardHeading';
```
Comment (danidewitt) [inline src/components/EzCard/EzCardHeading.tsx:42]: This seemed like the wrong display name, so I fixed it.

**Comment by danidewitt** [comment]
@CraigCav i think i've narrowed it down to the right components. let me know what you think?

**Review by CraigCav — APPROVED** [review]

**Comment by CraigCav** [comment]
Let's create a separate card for adding a lint rule to catch these going forwards.

**Pull request merged**